### ISSUE TITLE
Does not compile on Gradle 5+ (Unresolved Reference)

### ISSUE THREAD

**Issue by aphexcx** [onfido/onfido-android-sdk#48]
#### What was the expected behaviour?
Compiles and launches as it does on Gradle 4.10
#### What happened instead?
Compilation failed:
```
> Task :renter:compileStagingKotlin FAILED
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/OnfidoModule.kt: (4, 39): Unresolved reference: Onfido
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/OnfidoModule.kt: (5, 39): Unresolved reference: OnfidoFactory
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/OnfidoModule.kt: (13, 69): Unresolved reference: Onfido
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/OnfidoModule.kt: (14, 16): Unresolved reference: OnfidoFactory
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (27, 39): Unresolved reference: DocumentType
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (28, 39): Unresolved reference: ExitCode
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (29, 39): Unresolved reference: Onfido
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (30, 39): Unresolved reference: OnfidoConfig
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (31, 39): Unresolved reference: errors
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (32, 39): Unresolved reference: ui
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (33, 39): Unresolved reference: ui
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (34, 39): Unresolved reference: upload
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (35, 39): Unresolved reference: utils
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (36, 19): Unresolved reference: api
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (133, 40): Unresolved reference: Onfido
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (563, 78): Unresolved reference: Onfido
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (564, 21): 'onError' overrides nothing
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (564, 53): Unresolved reference: OnfidoException
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (564, 81): Unresolved reference: Applicant
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (568, 21): 'userCompleted' overrides nothing
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (568, 59): Unresolved reference: Applicant
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (568, 80): Unresolved reference: Captures
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (572, 21): 'userExited' overrides nothing
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (572, 55): Unresolved reference: ExitCode
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (572, 76): Unresolved reference: Applicant
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (750, 28): Unresolved reference: OnfidoConfig
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (754, 21): Unresolved reference: FlowStep
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (755, 21): Unresolved reference: CaptureScreenStep
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (755, 39): Unresolved reference: DocumentType
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (755, 69): Unresolved reference: CountryCode
e: /Users/afik/StudioProjects/skip-android/renter/src/main/java/com/waybots/ui/main/RenterActivity.kt: (755, 86): Unresolved reference: FlowStep

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':renter:compileStagingKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleKotlinCompilerWork
   > Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 17s
```

#### Version info:

- `onfido-capture-sdk` version: 4.2.0 (tried with 4.6.0 as well, same issue)
- Android API version(s): 28

- `compileSdkVersion`: 28
- `targetSdkVersion`: 28
- `minSdkVersion`: 19
- Support library version (if applicable):

#### Steps to reproduce:
Use this dependency: 
```
implementation "com.onfido.sdk.capture:onfido-capture-sdk:4.2.0"
```

With this gradle-wrapper.properties:
`distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-all.zip`

And this version of AGP:
`        classpath 'com.android.tools.build:gradle:3.5.0-alpha06'`

Using Android Studio 3.5 Canary 6

#### Onfido Flow initialization code: 
```
import android.content.Context
import com.onfido.android.sdk.capture.Onfido
import com.onfido.android.sdk.capture.OnfidoFactory
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class OnfidoModule {

    @Provides @Singleton fun provideOnfidoClient(context: Context): Onfido {
        return OnfidoFactory.create(context).client
    }
}
```

#### Other useful info:
_screenshots, videos, etc._

Note that the references are indeed not found by the IDE; this is how the OnfidoModule file looks:
![screen shot 2019-02-26 at 5 31 13 pm](https://user-images.githubusercontent.com/2283165/53458805-599c6e80-39ec-11e9-9541-f68fb83c487d.png)

### I noticed it builds and compiles fine _if I also add this dependency to my gradle file:_
`implementation "com.onfido.sdk.capture:onfido-capture-sdk-core:4.2.0"`

**Comment by shaubert** [comment]
+1

**Comment by andreluizreis** [comment]
I updated the Gradle today to the new version: classpath 'com.android.tools.build:gradle:3.4.0'
And it stopped working. I can't import the library anymore.

**Comment by freszu** [comment]
Just ran into this problem after gradle update using onfido 4.8.1. 

It seems that fix for this should be no more than one line - most likely the 
```com.onfido.sdk.capture:onfido-capture-sdk```
dependencies config is using ```implementation``` clause instead of ```api``` for core dependencies, please fix this. 

See:
https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation

**Comment by deividasstr** [comment]
Please fix, it has been a while. Many new features of android studio are enforcing android gradle plugin 3.4.x and gradle 5.x.x

**Comment by dmytririsov** [comment]
+1, need this fix too

**Comment by seewah** [comment]
Apologies for the inconvenience. We will look to fix this problem over then next few weeks.

**Comment by Moritz-Halfmeyer** [comment]
+1, same same

**Comment by SemenovAlexander** [comment]
@seewah any success on this? 
In our project, this issue started to add a lot of headaches with lint and proguard (we need to use external bouncy castle lib, which contains JPMS and can't update to Android plugin to 3.4 version because of this issue with Onfido):
https://stackoverflow.com/questions/54749277/unexpected-failure-during-lint-analysis-of-module-info-class

So we must disable lint check and also ignore proguard warnings because of this issue.

**Comment by sergiomarquesmoura** [comment]
Hi all,
Next week we will release a new version adding support for Gradle 5 and AS Gradle plugin 3.4+

We're just finishing testing some of our dependencies that required upgrades after this change and I'll update you all here once it's live.

Thanks for the patience and sorry for the inconvenience

**Comment by seewah** [comment]
Just a status update: unfortunately, there has been some delay in testing, and we don't want to release until we are 100% certain the version is good to go. Our new ETA for release is early next week (week commencing 15th July). Apologies for the delay

**Comment by deividasstr** [comment]
Hey @seewah, any updates?

**Comment by deividasstr** [comment]
This is becoming a pretty blocking issue for us @seewah @sergiomarquesmoura . At least an update on the situation would be helpful.

**Comment by Moritz-Halfmeyer** [comment]
> This is becoming a pretty blocking issue for us @seewah @sergiomarquesmoura . At least an update on the situation would be helpful.

Same for us. This is severely impacting our clients business currently. Please provide a new and accurate ETA this time.

**Comment by sergiomarquesmoura** [comment]
Hi all! We just released version `4.10.0` which supports solves this. Please bump the version and give us feedback. Full changelog here: https://github.com/onfido/onfido-android-sdk/releases/tag/4.10.0
